pages/pipelines/defining_steps.md.erb

```diff
@@ -270,6 +270,24 @@ steps:
 ```
 {: codeblock-file="pipeline.yml"}
 
+## Dynamic pipeline templates
+
+If you need the ability to use pipelines from a central catalog, or enforce certain configuration rules, you can either use dynamic pipelines and the [`pipeline upload`](/docs/agent/v3/cli-pipeline#uploading-pipelines) command to make this happen, or [write custom plugins](/docs/plugins/writing) and share them across your organization.
+
+To use dynamic pipelines and the pipeline upload command, you'd make a pipeline that looks something like this:
+
+```yml
+steps:
+  - command: enforce-rules.sh | buildkite-agent pipeline upload
+    label: "\:pipeline\: Upload"
+```
+
+Where each team defines their steps in `team-steps.yml`, and your templating logic is in `enforce-rules.sh` (which can be written in any language that can pass YAML to the pipeline upload).
+
+In `enforce-rules.sh` you can add steps to the YAML, require certain versions of dependencies or plugins, or implement any other logic you can program. Depending on your use case you might want to get `enforce-rules.sh` from an external catalog instead of committing it to the team repository.
```

Comment:
```suggestion
In `enforce-rules.sh` you can add steps to the YAML, require certain versions of dependencies or plugins, or implement any other logic you can program. Depending on your use case, you might want to get `enforce-rules.sh` from an external catalog instead of committing it to the team repository.
```